background.js

```diff
@@ -1,128 +1,374 @@
-}
-
+/**
+ * Version info
+ *
+ */
+
+chrome.browserAction.onClicked.addListener(rcxMain.inlineToggle);
+chrome.tabs.onSelectionChanged.addListener(rcxMain.onTabSelect);
+chrome.runtime.onMessage.addListener(
+	function (request, sender, response) {
+		switch (request.type) {
+			case 'enable?':
+				console.log('enable?');
+				rcxMain.onTabSelect(sender.tab.id);
+				break;
+			case 'xsearch':
+				console.log('xsearch');
+				var e = rcxMain.search(request.text, request.dictOption);
+				response(e);
+				break;
+			/**  case 'nextDict':
+			 console.log('nextDict');
+			 rcxMain.nextDict();
+			 break;*/
+			case 'resetDict':
+				console.log('resetDict');
+				rcxMain.resetDict();
+				break;
+			case 'translate':
+				console.log('translate');
+				var e = rcxMain.dict.translate(request.title);
+				response(e);
+				break;
+			case 'makehtml':
+				console.log('makehtml');
+				var html = rcxMain.dict.makeHtml(request.entry);
+				response(html);
+				break;
+			case 'switchOnlyReading':
+				console.log('switchOnlyReading');
+				if (rcxMain.config.onlyreading == 'true')
+					rcxMain.config.onlyreading = 'false';
+				else
+					rcxMain.config.onlyreading = 'true';
+				//Todo: Replace this with cloud storage
+				localStorage['onlyreading'] = rcxMain.config.onlyreading;
+				break;
+			case 'copyToClip':
+				console.log('copyToClip');
+				rcxMain.copyToClip(sender.tab, request.entry);
+				break;
+			default:
+				console.log(request);
+		}
+	})
+
+var optionsList = [
+	"copySeparator", // Is this field separator?
+	"disablekeys",
+	"highlight",
+	"kanjicomponents",
+	"kanjiInfo",
+	"lineEnding",
+	"maxClipCopyEntries",
+	"minihelp",
+	"onlyreading",
+	"popupcolor",
+	"popupDelay",
+	"popupLocation",
+	"textboxhl",
+	"showOnKey"];
+
+var kanjiInfoLabelList = rcxDict.prototype.kanjiInfoLabelList;
+this.kanjiInfoObject = {}; //Object used to save kanjiinfo to cloud.
+rcxMain.config = {};
+
+/** Get option data from cloud. */
+chrome.storage.sync.get(optionsList,
+	function (items) {
+		fillRcxValsFromLocalStoreCloudAndDefaultValues(items);
+		saveOptionsToCloudStorage();
+	})
+
+
+/**
+ * Initializes the localStorage for the given key.
+ * If the given key is already initialized, nothing happens.
+ *
+ * @author Teo (GD API Guru)
+ * @param key The key for which to initialize
+ * @param initialValue Initial value of localStorage on the given key
+ * @return true if a value is assigned or false if nothing happens
+ */
+function initStorage(key, initialValue) {
+	var currentValue = localStorage[key];
+	if (!currentValue) {
+		localStorage[key] = initialValue;
+		return true;
+	}
+	return false;
+}
+
+/**
+ * Populates config with option values from 3 places
+ * 1. Cloud Storage, 2. Local Storage, and 3. Default
+ * If any values are in local storage they will overwrite cloud storage values.
+ * Values should no longer be saved to local storage.
+ *
+ * @author Deshaun (From Dev Color)
```

Comment:
you don't actually need this since you're submitting directly to the repro and so your authorship is embedded in metadata (unlike above where I just took code from web)